src/make_watertight/Arc.cpp

```diff
@@ -0,0 +1,713 @@
+    assert(moab::MB_SUCCESS == result);
+    if (2 != end_verts.size()) {
+      std::cout << "end_verts.size()=" << end_verts.size() << std::endl;
+      gen->print_edge(edge);
+    }
+    assert(2 == end_verts.size());
+    vert = end_verts.front() == vert ? end_verts.back() : end_verts.front();
+    ordered_verts.push_back(vert);
+  }
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Arc::get_meshset(const moab::EntityHandle set, std::vector<moab::EntityHandle>& vec) {
+  moab::ErrorCode result;
+  vec.clear();
+  result = MBI()->get_entities_by_handle(set, vec);
+  assert(moab::MB_SUCCESS == result);
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Arc::set_meshset(const moab::EntityHandle set, const std::vector<moab::EntityHandle> vec) {
+  moab::ErrorCode result;
+  result = MBI()->clear_meshset(&set, 1);
+  assert(moab::MB_SUCCESS == result);
+  result = MBI()->add_entities(set, &vec[0], vec.size());
+  assert(moab::MB_SUCCESS == result);
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Arc::merge_verts(const moab::EntityHandle keep_vert,
+                                 const moab::EntityHandle delete_vert,
+                                 std::vector<moab::EntityHandle>& arc0,
+                                 std::vector<moab::EntityHandle>& arc1) {
+
+  moab::ErrorCode rval;
+  // first update the arcs with the keep_vert
+  for (std::vector<moab::EntityHandle>::iterator i = arc0.begin(); i != arc0.end(); ++i) {
+    if (delete_vert == *i)
+      *i = keep_vert;
+  }
+  for (std::vector<moab::EntityHandle>::iterator i = arc1.begin(); i != arc1.end(); ++i) {
+    if (delete_vert == *i)
+      *i = keep_vert;
+  }
+
+  // get adjacent tris
+  moab::Range tris;
+  moab::EntityHandle verts[2] = {keep_vert, delete_vert};
+  rval = MBI()->get_adjacencies(verts, 2, 2, false, tris, moab::Interface::UNION);
+  MB_CHK_SET_ERR(rval, "getting adjacent tris failed");
+
+  // actually do the merge
+  rval = MBI()->merge_entities(keep_vert, delete_vert, false, true);
+  MB_CHK_SET_ERR(rval, "merge entities failed");
+
+  // delete degenerate tris
+  rval = zip->delete_degenerate_tris(tris);
+  MB_CHK_SET_ERR(rval, "deleting degenerate tris failed");
+
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Arc::merge_curves(moab::Range curve_sets, const double facet_tol,
+                                  moab::Tag id_tag, moab::Tag merge_tag, const bool debug) {
+  // find curve endpoints to add to kd tree
+  moab::ErrorCode result;
+  const double SQR_TOL = facet_tol * facet_tol;
+  moab::Range endpoints;
+  for (moab::Range::iterator i = curve_sets.begin(); i != curve_sets.end(); i++) {
+    std::vector<moab::EntityHandle> curve;
+    result = get_meshset(*i, curve);
+    assert(moab::MB_SUCCESS == result);
+    assert(1 < curve.size());
+    moab::EntityHandle front_endpt = curve[0];
+    moab::EntityHandle back_endpt  = curve[curve.size() - 1];
+    // ADD CODE TO HANDLE SPECIAL CASES!!
+    if (front_endpt == back_endpt) { // special case
```

Comment:
Error Handling: Until this special case is implemented, you can print a warning within these `if`/`else` blocks.  This lets the user know that a special case was hit, but has yet to be implemented.